src/Compilers/CSharp/Portable/Symbols/TypeParameterBounds.cs

```diff
@@ -92,19 +97,23 @@ internal static bool HasValue(this TypeParameterBounds? boundsOpt, bool canIgnor
             {
                 return true;
             }
-            return canIgnoreNullableContext || !boundsOpt.IgnoresNullableContext;
+            return canUseLightweightTypeConstraintBinding || !boundsOpt.UsedLightweightTypeConstraintBinding;
         }
 
-        // Returns true if bounds was updated with value.
-        // Returns false if bounds already had a value with sufficient 'IgnoresNullableContext'
-        // or was updated to a value with sufficient 'IgnoresNullableContext' on another thread.
+        /// <summary>
+        /// Returns true if bounds was updated with value.
+        /// Returns false if bounds already had a value with sufficient 'canUseLightweightTypeConstraintBinding'
```

Comment:
>sufficient 'canUseLightweightTypeConstraintBinding' [](start = 61, length = 51)

What does `sufficient` mean?

Reply:
If someone asks for fully-bound info, we cannot return light-bound info, it is not sufficient.

---
In reply to: [502588978](https://github.com/dotnet/roslyn/pull/48270#discussion_r502588978) [](ancestors = 502588978)